src/mcp/types.py

```diff
@@ -64,8 +64,10 @@ class Meta(BaseModel):
     """
 
 
-RequestParamsT = TypeVar("RequestParamsT", bound=RequestParams)
-NotificationParamsT = TypeVar("NotificationParamsT", bound=NotificationParams)
+RequestParamsT = TypeVar("RequestParamsT", bound=RequestParams | dict[str, Any] | None)
+NotificationParamsT = TypeVar(
+    "NotificationParamsT", bound=NotificationParams | dict[str, Any] | None
+)
```

Comment:
This is a bit weird tho... I'd like to spend some minutes checking if we can be a bit stricter here.

Reply:
I'll do it tomorrow.